aws/table_aws_ssoadmin_permission_set.go

```diff
@@ -0,0 +1,241 @@
+				Description: "The name of the permission set.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSsoAdminPermissionSet,
+				Transform:   transform.FromField("PermissionSet.Name"),
+			},
+			{
+				Name:        "arn",
+				Description: "The ARN of the permission set.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("PermissionSetArn"),
+			},
+			{
+				Name:        "created_date",
+				Description: "The date that the permission set was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getSsoAdminPermissionSet,
+				Transform:   transform.FromField("PermissionSet.CreatedDate"),
+			},
+			{
+				Name:        "description",
+				Description: "The description of the permission set.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSsoAdminPermissionSet,
+				Transform:   transform.FromField("PermissionSet.Description"),
+			},
+			{
+				Name:        "instance_arn",
+				Description: "The Amazon Resource Name (ARN) of the SSO Instance under which the operation will be executed.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "relay_state",
+				Description: "Used to redirect users within the application during the federation authentication process.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSsoAdminPermissionSet,
+				Transform:   transform.FromField("PermissionSet.RelayState"),
+			},
+			{
+				Name:        "session_duration",
+				Description: "The length of time that the application user sessions are valid for in the ISO-8601 standard.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSsoAdminPermissionSet,
+				Transform:   transform.FromField("PermissionSet.SessionDuration"),
+			},
+			{
+				Name:      "tags_src",
+				Type:      proto.ColumnType_JSON,
+				Hydrate:   getSsoAdminPermissionSetTags,
+				Transform: transform.FromValue(),
+			},
+
+			// Standard columns for all tables
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getSsoAdminPermissionSetTags,
+				Transform:   transform.From(getSsoAdminResourceTurbotTags),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getSsoAdminPermissionSet,
+				Transform:   transform.FromField("PermissionSet.Name"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("PermissionSetArn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listSsoAdminPermissionSets(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listSsoAdminPermissionSets")
+
+	instanceArn := d.KeyColumnQuals["instance_arn"].GetStringValue()
+
```

Comment:
Hi @LalitTurbot . I tried all of the queries you suggested, but they all work:

```
> select name, ps.arn, created_date, description, relay_state, session_duration, tags from aws.aws_ssoadmin_permission_set as ps join aws.aws_ssoadmin_instance as i on ps.instance_arn = i.arn
+---------+-------------------------------------------------------------------------+---------------------+-------------------+-------------------------+------------------+--------------------+
| name    | arn                                                                     | created_date        | description       | relay_state             | session_duration | tags               |
+---------+-------------------------------------------------------------------------+---------------------+-------------------+-------------------------+------------------+--------------------+
| Billing | arn:aws:sso:::permissionSet/ssoins-xxxxxxxxxxxxxxxx/ps-xxxxxxxxxxxxxxxx | 2021-09-13 07:15:16 | Access to billing | http://www.example.com/ | PT1H             | {"Type":"Billing"} |
+---------+-------------------------------------------------------------------------+---------------------+-------------------+-------------------------+------------------+--------------------+

> select * from aws_ssoadmin_instance
+------------------------------------------------+-------------------+------------------------------------------------+----------------------------------------------------+-----------+-----------+--------------+
| arn                                            | identity_store_id | title                                          | akas                                               | partition | region    | account_id   |
+------------------------------------------------+-------------------+------------------------------------------------+----------------------------------------------------+-----------+-----------+--------------+
| arn:aws:sso:::instance/ssoins-xxxxxxxxxxxxxxxx | d-xxxxxxxxxx      | arn:aws:sso:::instance/ssoins-xxxxxxxxxxxxxxxx | ["arn:aws:sso:::instance/ssoins-xxxxxxxxxxxxxxxx"] | aws       | us-east-1 | 799416384551 |
+------------------------------------------------+-------------------+------------------------------------------------+----------------------------------------------------+-----------+-----------+--------------+

> select * from aws_ssoadmin_permission_set where instance_arn = 'arn:aws:sso:::instance/ssoins-xxxxxxxxxxxxxxxx'
+---------+-------------------------------------------------------------------------+---------------------+-------------------+------------------------------------------------+-------------------------+----------------
| name    | arn                                                                     | created_date        | description       | instance_arn                                   | relay_state             | session_duratio
+---------+-------------------------------------------------------------------------+---------------------+-------------------+------------------------------------------------+-------------------------+----------------
| Billing | arn:aws:sso:::permissionSet/ssoins-xxxxxxxxxxxxxxxx/ps-xxxxxxxxxxxxxxxx | 2021-09-13 07:15:16 | Access to billing | arn:aws:sso:::instance/ssoins-xxxxxxxxxxxxxxxx | http://www.example.com/ | PT1H           
+---------+-------------------------------------------------------------------------+---------------------+-------------------+------------------------------------------------+-------------------------+----------------
(END)
```

The only time I got the errors you saw were when I encountered this issue: https://github.com/turbot/steampipe-plugin-sdk/issues/182

I think a way of resolving the issues that we're having would be to make `instance_arn` optional - @e-gineer mentioned that typically the issues I saw happen when there are multiple key columns. Given that an AWS account only allows a single `aws_ssoadmin_instance` to be present at a time I can look up the `instance_arn` and only need a single required key column. It would also make it easier to use this table. @LalitTurbot / @e-gineer - would that seem like a reasonable option?

Reply:
The other cheat would be to parse the instance ID from the passed in permission set ARN and build a new ARN from that. E.g. parse out:

arn:aws:sso:::permissionSet/**ssoins-1234567890abcdef**/ps-xxxxxxxxxxxxxxxx

Query for instance ARN:

arn:aws:sso:::instance/**ssoins-1234567890abcdef**

Which makes you wonder why AWS asks you to pass the instance ARN as part of the call to DescribePermissionSet, but given the inconsistent state of the tagging APIs wierdness in other APIs isn't unusual...